### PR TITLE
engine/schema: move SQLs to 4.20.0 to 4.20.1 upgrade

### PR DESCRIPTION
**Pull request by weizhouapache**
### Description

This PR 
- adds the upgrade path for 4.20.1
- move some new SQLs from 4.19.1->4.20.0 to 4.20.0->4.20.1

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI
- [ ] test (unit or integration test code)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by weizhouapache** [comment]
@blueorangutan package

**Review by shwstppr — APPROVED** [review]
Code lgtm

**Comment by weizhouapache** [comment]
@blueorangutan test

**Comment by weizhouapache** [comment]
double-checked the database, looks ok

```
mysql> select * from version;
...
| 45 | 4.20.0.0 | 2024-12-02 10:04:53 | Complete |
| 46 | 4.20.1.0 | 2024-12-02 10:04:53 | Complete |
+----+----------+---------------------+----------+


mysql> show create table mshost_peer;
...
| mshost_peer | CREATE TABLE `mshost_peer` (
  `id` bigint unsigned NOT NULL AUTO_INCREMENT,
  `owner_mshost` bigint unsigned NOT NULL,
  `peer_mshost` bigint unsigned NOT NULL,
  `peer_runid` bigint NOT NULL,
  `peer_state` varchar(10) NOT NULL DEFAULT 'Down',
  `last_update` datetime DEFAULT NULL COMMENT 'Last record update time',
  PRIMARY KEY (`id`),
  UNIQUE KEY `i_mshost_peer__owner_peer` (`owner_mshost`,`peer_mshost`),
  KEY `fk_mshost_peer__peer_mshost` (`peer_mshost`),
  CONSTRAINT `fk_mshost_peer__owner_mshost` FOREIGN KEY (`owner_mshost`) REFERENCES `mshost` (`id`) ON DELETE CASCADE,
  CONSTRAINT `fk_mshost_peer__peer_mshost` FOREIGN KEY (`peer_mshost`) REFERENCES `mshost` (`id`)
) ENGINE=InnoDB AUTO_INCREMENT=4 DEFAULT CHARSET=utf8mb3 |
```

**Review by sureshanaparti — APPROVED** [review]
clgtm

**Review by bernardodemarco — APPROVED** [review]
CLGTM

File: engine/schema/src/main/java/com/cloud/upgrade/dao/Upgrade42000to42010.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+// specific language governing permissions and limitations
+// under the License.
+package com.cloud.upgrade.dao;
+
+import com.cloud.utils.exception.CloudRuntimeException;
+
+import java.io.InputStream;
+import java.sql.Connection;
+
+public class Upgrade42000to42010 implements DbUpgrade {
```
[the remark this answers is not in the document]
Comment (weizhouapache) [reply engine/schema/src/main/java/com/cloud/upgrade/dao/Upgrade42000to42010.java:24]: @rohityadavcloud 
good point

currently it is unnecessary, as we do not have new systemvm template for 4.20.1, so the 4.20.0.0 systemvm template will be seeded when upgrade from 4.19 and elder versions.
```
2024-12-02 10:22:14,754 DEBUG [c.c.u.s.Script] (qtp253011924-19:[ctx-adaef633, ctx-64361295, ctx-7804f1df]) (logid:86c544ef) Absolute path =  /usr/share/cloudstack-common/scripts/storage/secondary/setup-sysvm-tmplt
2024-12-02 10:22:14,754 DEBUG [c.c.u.SystemVmTemplateRegistration] (qtp253011924-19:[ctx-adaef633, ctx-64361295, ctx-7804f1df]) (logid:86c544ef) Executing command [/usr/share/cloudstack-common/scripts/storage/secondary/setup-sysvm-tmplt -u 250d238a-5140-4ba3-9b59-ae7324430dc3 -f /usr/share/cloudstack-management/templates/systemvm/systemvmtemplate-4.20.0-x86_64-kvm.qcow2.bz2 -h kvm -d /tmp/tmp6751224453432784328/template/tmpl/1/3 ].
```

we can add it when we build new systemvm template

**Comment by rohityadavcloud** [comment]
@weizhouapache my bad, looks like I merged https://github.com/apache/cloudstack/commit/d17de834a57237b79f24850f97190bfc378f31a9 which already created an upgrade path. Please check and close or fix conflicts on this PR.

**Comment by weizhouapache** [comment]
> @weizhouapache my bad, looks like I merged [d17de83](https://github.com/apache/cloudstack/commit/d17de834a57237b79f24850f97190bfc378f31a9) which already created an upgrade path. Please check and close or fix conflicts on this PR.

@rohityadavcloud 
it is fine, no worries, closing this PR

**Comment by weizhouapache** [comment]
> > @weizhouapache my bad, looks like I merged [d17de83](https://github.com/apache/cloudstack/commit/d17de834a57237b79f24850f97190bfc378f31a9) which already created an upgrade path. Please check and close or fix conflicts on this PR.
> 
> @rohityadavcloud it is fine, no worries, closing this PR

oh, my bad, we need to move some SQLs as well

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan test

**Pull request merged**